compiler/ilgen/IlBuilder.cpp

```diff
@@ -1419,6 +1419,53 @@ IlBuilder::Call(const char *functionName, int32_t numArgs, TR::IlValue ** argVal
    return NULL;
    }
 
+/**
+ * This service is created for incrementing the value
+ * of address+offset postion so the input "address"
+ * has to be TR::Address type.
+ * atomicAdd32BitSymbol can generate "lock add" in codegen.
+ * This service is specifically for Int32.
+ */
+TR::IlValue *
+IlBuilder::AtomicInt32AddWithOffset(TR::IlValue * address, TR::IlValue * offset, TR::IlValue * value)
+   {
+   TR_ASSERT(address->getSymbol()->getDataType() == TR::Address || offset->getSymbol()->getDataType() == TR::Address, "The IlType of first two operands must be TR::Address");
+   TR_ASSERT(value->getSymbol()->getDataType() == TR::Int32, "Thle IlType of value must be TR::Int32");
+   TraceIL("IlBuilder[ %p ]::AtomicInt32AddWithOffset (%d, %d, %d)\n", this, address->getCPIndex(), offset==NULL?0:offset->getCPIndex(), value->getCPIndex());
+   appendBlock();
+
+   TR::DataType returnType = TR::Int32;
+   
+   TR::SymbolReference *methodSymRef = symRefTab()->findOrCreateCodeGenInlinedHelper(TR::SymbolReferenceTable::atomicAdd32BitSymbol); //lock add
+   TR::Node *callNode;
+   callNode = TR::Node::createWithSymRef(TR::ILOpCode::getDirectCall(returnType), offset==NULL?2:3, methodSymRef);
+   callNode->setAndIncChild(0, loadValue(address));
+   if(offset==NULL){
+      callNode->setAndIncChild(1, loadValue(value));
+   }
+   else{
```

Comment:
Just pushed another version. 1) handled the offset is NULL issue  2) changed code style consistent with others  3) change the format of comments (thanks to Yi).